### ISSUE TITLE
Please support AUR packages

### ISSUE THREAD

**Issue by IPlayZed** [PackageKit/PackageKit#704]
Reiterating on #340.

Yes, I use Flatpaks mainly, but there are several AUR packages which provide the system functionality I need, like game-devices-udev for Steam.
It would be great if I didn't have to run an AUR helper manually each time.

Reopening #340 or at least finding someone who has the knowledge to implement this would be great, instead of closing this forcefully.

Mostly we use wrappers like paru or yay anyway, is there some functionality not present there which would not allow them to be used instead of pacman directly? Arch packages use the same build system anyway.

**Comment by IPlayZed** [comment]
So, if there is something option to always accept changes, it should work fine?
I am questioning how other hooks are handled then, regardless of AUR support.
For instance informant.
If there is no interactivity support, on the long term it might be a good idea to add it.

**Comment by ximion** [comment]
This has been a matter of debate for a long time (known as hughsie's law), and it will not change. Installations, updates and removals must run without users interaction, the only exception being displayed EULA on installation time.

Even if we wanted to, at this point we could not change it as all PackageKit clients expect this behavior. Especially if you are running offline-updates, you can not really have user interactions.

**Comment by IPlayZed** [comment]
> Especially if you are running offline-updates, you can not really have user interactions.
Doesn't SystemD and Plymouth provide user interaction, even in a graphical manner?
But, if I understand correctly as long there was a method worked out so that AUR helpers can work in a fire and forget mode, this would be doable?

**Comment by ximion** [comment]
Yes, if there is no interactivity beyond "transaction failed" with an error message, everything is fine!
On Debian, we bolted on a tiny bit of interactivity for certain installations, but that is not a universal system and currently most of the time frontends run stuff in fully noninteractive mode anyway.

PackageKit is designed for users who either have no idea how a computer works, so can't make any sensible policy decision, or for people who do know that but don't want to deal with the computer nagging constantly and just want it to perform the task with the best choices.

So, if the alpm backend can just install AUR stuff noninteractively and keep it up-to-date like that too, there's no reason why that couldn't be merged :-)

**Comment by IPlayZed** [comment]
@ximion It seems like [wrappers can do operations unattended](https://www.reddit.com/r/archlinux/comments/nbsefm/does_the_paru_aur_helper_have_a_similar_option_to/).

I am not at all knowledgeable about PackageKit's workings, does it use libalpm bindings or something from code directly? If it uses Pacman directly, changing the command to be wrapped to a given executable like Paru or Yay, would solve the issue I think with the correct option passed, as written in the post. If it uses libalpm, I could open an issue in the one of the AUR managers' project pages asking how to proceed.

**Comment by Lazerbeak12345** [comment]
As a note, in case this is closed as a duplicate prematurely with the linked issue unopened, packagekit supports COPR and PPA repos. The AUR is nearly identical to these. If you don't want AUR support, and you can justify it, then one can apply the same justification to argue for removing COPR and PPA - said as someone who'd prefer support for all of the mentioned.

**Comment by ximion** [comment]
@IPlayZed PackageKit uses backends maintained by the respective distributions to do any package-manager specific work. For Arch, this is the `alpm` backend, which uses libalpm to do its work.
Any AUR support would have to be implemented here, and as long as unattended installation is possible, there's no issue with having it (you'd have to have a smart implementation for it, maybe exposing it as repository within PackageKit).

I am not knowledgeable enough about Arch though to give much further advice on the specifics. 
You can find the alpm backend code here: https://github.com/PackageKit/PackageKit/tree/main/backends/alpm
You might want to ask one of its authors for advice: https://github.com/PackageKit/PackageKit/blob/main/backends/alpm/pk-backend-alpm.c#L48
